// @capgo/capacitor-media-session/dist/plugin.js
var capacitorMediaSession = (function (exports, core) {
    'use strict';

    const MediaSession = core.registerPlugin('MediaSession', {
        web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.MediaSessionWeb()),
        ios: () => Promise.resolve().then(function () { return web; }).then((m) => new m.MediaSessionWeb()),
    });

    class MediaSessionWeb extends core.WebPlugin {
        async setMetadata(options) {
            const mediaSession = navigator.mediaSession;
            if (!mediaSession) {
                throw this.unavailable('Media Session API not available in this browser.');
            }
            const MediaMetadataCtor = window.MediaMetadata;
            if (MediaMetadataCtor) {
                mediaSession.metadata = new MediaMetadataCtor(options);
            }
            else {
                mediaSession.metadata = options;
            }
        }
        async setPlaybackState(options) {
            const mediaSession = navigator.mediaSession;
            if (!mediaSession) {
                throw this.unavailable('Media Session API not available in this browser.');
            }
            mediaSession.playbackState = options.playbackState;
        }
        async setActionHandler(options, handler) {
            const mediaSession = navigator.mediaSession;
            if (!mediaSession) {
                throw this.unavailable('Media Session API not available in this browser.');
            }
            mediaSession.setActionHandler(options.action, handler);
        }
        async setPositionState(options) {
            const mediaSession = navigator.mediaSession;
            if (!mediaSession) {
                throw this.unavailable('Media Session API not available in this browser.');
            }
            mediaSession.setPositionState(options);
        }
        async getPluginVersion() {
            return { version: 'web' };
        }
    }

    var web = /*#__PURE__*/Object.freeze({
        __proto__: null,
        MediaSessionWeb: MediaSessionWeb
    });

    exports.MediaSession = MediaSession;

    return exports;

})({}, capacitorExports);
